test/index.js

```diff
@@ -60,4 +76,102 @@ describe('Schema#virtual()', function () {
     assert.equal('object', typeof model.populatedVirtuals);
     assert.deepEqual(model.populatedVirtuals.foobar, { _id: 3 });
   });
+
+  //it('throws error if populating undefined virtual', function (done) {
+  //  var schema = new mongoose.Schema({
+  //    name: { type: String},
+  //  });
+  //
+  //  schema.virtual('foobar', {
+  //    ref: 'Foobar',
+  //    localKey: 'foobarId',
+  //    foreignKey: '_id',
+  //  });
+  //
+  //  var Model = mongoose.model('Model', schema);
+  //
+  //  var m = new Model({name: 'model'});
+  //  m.save();
+  //
+  //  //TODO add assert
+  //  Model
+  //    .findOne()
+  //    .populate('foobar')
+  //    .exec(function (err, model) {
+  //      if (err) {
+  //        console.log(err, err.stack);
+  //        done(err);
+  //      } else {
+  //        console.log('model async: ', model);
+  //        done();
+  //      }
+  //    });
+  //});
+
+  it('should populate correctly', function (done) {
+    mongoose = populateVirtuals(mongoose);
+
+    var bookSchema = new mongoose.Schema({
+      title: { type: String },
+      authorId: { type: mongoose.Schema.Types.ObjectId, ref: 'Author'},
```

Comment:
This plugin changes `.populate()` functionality to populate virtuals instead of objectId fields. This will not work.